src/service/translate.js

```diff
@@ -1780,10 +1780,10 @@ function $translate($STORAGE_KEY, $windowProvider, $translateSanitizationProvide
           // we are already loading this asynchronously
           // resolve our new deferred when the old langPromise is resolved
           langPromises[key].then(function (translation) {
-            if (!$uses) {
-              useLanguage(translation.key);
+            if ($nextLang === key) {
```

Comment:
Just took it as it was below - could we have the situation where key is already running out if translation.key? Maybe I was just misreading it.